pkg/operator/rules.go

```diff
@@ -25,9 +25,31 @@ import (
 	"github.com/prometheus/prometheus/model/rulefmt"
 
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/informers"
+	"github.com/prometheus-operator/prometheus-operator/pkg/k8sutil"
+	namespacelabeler "github.com/prometheus-operator/prometheus-operator/pkg/namespace-labeler"
 	thanostypes "github.com/thanos-io/thanos/pkg/store/storepb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type PrometheusRuleSelector struct {
+	objectType   string
```

Comment:
I see we still need this field for partial response strategy field

Reply:
We can make it something more obvious though

```suggestion
	ruleFormat   RuleConfigurationFormat
```

with

```golang
type RuleConfigurationFormat int

const (
    PrometheusFormat RuleConfigurationFormat = iota
    ThanosFormat
)
```